R/prevalence-DELTA_plots.R

```diff
@@ -601,73 +722,129 @@ forest_delta_plotly <- function(
     arrow_lwd     = 0.6,
     arrow_head_cm = 0.30,
     show_grid     = FALSE,
-    # NEW:
     base_text_pt  = 12,
     font_family   = "Montserrat",
     seg_width     = 1.6,
-    point_size    = 11
+    point_size    = 11,
+    stat          = c("T", "Z")
 ) {
   if (!is.data.frame(x)) stop("`x` must be a data.frame/tibble.")
-  need_cols <- c("rank","feature","group1","group2","design","T_obs","p_perm","p_adj_rank","category_rank_bh")
+  stat <- match.arg(stat)
+
+  need_cols <- c("rank","feature","group1","group2","design",
+                 "T_obs","p_perm","p_adj_rank","category_rank_bh")
   miss <- setdiff(need_cols, colnames(x))
   if (length(miss)) stop(paste("`x` is missing required columns:", paste(miss, collapse = ", ")))
   if (!is.null(n_each)) { n_neg_each <- n_pos_each <- as.integer(n_each) }
 
-  if (isTRUE(add_signed_z_from_p)) {
-    x$Z_signed_from_p_2s <- sign(x$T_obs) * stats::qnorm(1 - x$p_perm/2)
+  # compute signed Z from p if requested or needed for plotting
+  if (isTRUE(add_signed_z_from_p) || identical(stat, "Z")) {
+    if (!"p_perm" %in% names(x)) {
+      stop("Column `p_perm` is required to compute signed Z from p-values.")
+    }
+    p_two <- x$p_perm
+    p_two[!is.na(p_two)] <- pmin(
+      pmax(p_two[!is.na(p_two)], .Machine$double.eps),
+      1 - .Machine$double.eps
+    )
+    x$Z_signed_from_p_2s <- sign(x$T_obs) * stats::qnorm(1 - p_two / 2)
```

Comment:
This is duplicated code. Maybe extract to small helper function (as we can see, this content changes ;)